typescript/core/src/McapStreamReader.ts

```diff
@@ -92,27 +92,42 @@ export default class McapStreamReader {
       throw new Error("Already done reading");
     }
     this.#appendOrShift(data);
-    this.#reader.reset(this.#view);
   }
   #appendOrShift(data: Uint8Array): void {
     /** Add data to the buffer, shifting existing data or reallocating if necessary. */
-    const remainingBytes = this.#reader.bytesRemaining();
+    const consumedBytes = this.#reader.offset;
+    const remainingBytes = this.#view.byteLength - consumedBytes;
     const totalNeededBytes = remainingBytes + data.byteLength;
 
     if (totalNeededBytes <= this.#buffer.byteLength) {
       // Data fits in the current buffer
-      if (this.#view.byteOffset + totalNeededBytes <= this.#buffer.byteLength) {
+      if (
+        this.#view.byteOffset + this.#view.byteLength + data.byteLength <=
```

Comment:
To check if the new data can be appended (without shifting existing data), we do not care about how many bytes the reader has consumed. This was the case before, where `totalNeededBytes` was used.

Reply:
```
this.#view.byteLength + data.byteLength != totalNeededBytes
this.#view.byteLength + data.byteLength != this.#view.byteLength - consumedBytes + data.byteLength;
this.#view.byteLength != this.#view.byteLength - consumedBytes;
```